src/librustdoc/html/static/js/search.js

```diff
@@ -1623,10 +1656,20 @@ window.initSearch = function(rawSearchIndex) {
 
     searchWords = buildIndex(rawSearchIndex);
     registerSearchEvents();
-    // If there's a search term in the URL, execute the search now.
-    if (searchState.getQueryStringParams().search) {
-        search();
+
+    function runSearchIfNeeded() {
+        var searchParams = searchState.getQueryStringParams();
+
+        if (searchParams["filter-crate"] !== undefined) {
+            updateLocalStorage("saved-filter-crate", searchParams["filter-crate"]);
+        }
```

Comment:
We should not update the stored setting unless the user specifically made a selection from the dropdown. For instance, if I send you a link that includes a `&crate=foo` parameter, and you click on it, that shouldn't affect your saved settings.

Reply:
Removed all local storage for this parameter.